content/learning-paths/cross-platform/kleidiai-explainer/page3.md

```diff
@@ -0,0 +1,256 @@
+---
+title: Quantizing and packing micro-kernels
+weight: 4
+
+### FIXED, DO NOT MODIFY
+layout: learningpathall
+---
+
+## KleidiAI GitLab Repo
+This section will probe into the intricate details of what KlediAI is doing and how it is beneficial to AI performance optimization. Navigate to the [KlediAI GitLab repository](https://gitlab.arm.com/kleidi/kleidiai) and view the [example file](https://gitlab.arm.com/kleidi/kleidiai/-/blob/main/examples/matmul_clamp_f32_qai8dxp_qsi4cxp/matmul_clamp_f32_qai8dxp_qsi4cxp.cpp?ref_type=heads) you will be running. This example highlights the *i8mm* matrix multiplication micro-kernel, alongside the enabling packing/quantization micro-kernels.
+
+**Note:** This example is intended to illustrite KleidiAI microkernel performance. In practice you will not interact with KleidiAI's microkernels as your ML framework will leverage it automatically if supported. 
+
+The example code is structured to take in two matricies and compare KleidiAI's micro-kernel results to a reference implementation. This reference code is functionally identical to KleidiAI's micro-kernels, and is present for two reasons:
+1. To check for KleidiAI micro-kernel output validity as a sanity check, and
+2. To explain how KleidiAI micro-kernels work. The implementation within KleidiAI's micro-kernels use hand-optimized C and Arm Assembly code, naturally being difficult to interpret.
+
+
+## Build and Run the KleidiAI C++ example (i8mm micro-kernel)
+
+Follow these steps to build and run the KleidiAI library and example script reviewing the:
+
+1. Create an Ubuntu 24 Arm Linux machine on an AWS EC2 instance. Use an M7g-medium, which uses the Graviton 3 SoC supporting the *i8mm* Arm architecture feature. The 1 CPU and 4 GB of RAM in the M7g-medium are sufficient for this basic example run.
```

Comment:
Feels like Ubuntu is usually referred to with the full name, like Ubuntu 24.04. Would be nice to refer some AWS LP here as well

Reply:
Done for both